From c7cd7ba58bdcd336de3caf99e14fa8c225c76808 Mon Sep 17 00:00:00 2001
From: Peter Andreas Entschev <peter@entschev.com>
Date: Fri, 8 May 2026 02:34:00 -0700
Subject: [PATCH] Allow overriding sysusers install directory

The systemd pkg-config file reports an absolute sysusers.d directory
outside conda's install prefix. Make the directory configurable so the
feedstock can keep installed files inside the package prefix.
---
 CMakeLists.txt | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6c923b68a..5f4671b89 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -600,11 +600,13 @@ RDMA_DoFixup("${SYSTEMD_FOUND}" "systemd/sd-daemon.h")
 
 # https://www.freedesktop.org/software/systemd/man/latest/sysusers.d.html
-set(SYSUSERS_DIR "")
-if (NOT ${CMAKE_VERSION} VERSION_LESS "3.4")
-  pkg_check_modules(SYSTEMD_PC systemd QUIET)
-  if (SYSTEMD_PC_FOUND)
-    pkg_get_variable(SYSUSERS_DIR systemd sysusersdir)
+set(SYSUSERS_DIR "" CACHE PATH "Location for systemd sysusers.d files")
+if ("${SYSUSERS_DIR}" STREQUAL "")
+  if (NOT ${CMAKE_VERSION} VERSION_LESS "3.4")
+    pkg_check_modules(SYSTEMD_PC systemd QUIET)
+    if (SYSTEMD_PC_FOUND)
+      pkg_get_variable(SYSUSERS_DIR systemd sysusersdir)
+    endif()
   endif()
 endif()
 if ("${SYSUSERS_DIR}" STREQUAL "")
   set(SYSUSERS_DIR "/usr/lib/sysusers.d")
-- 
2.40.1
